akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedStashReferenceManagement.scala

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.internal
+
+import akka.actor.typed.scaladsl.StashBuffer
+import akka.annotation.InternalApi
+import akka.persistence.typed.internal.EventsourcedBehavior.InternalProtocol
+
+@InternalApi
+trait EventsourcedStashReferenceManagement {
+
+  private var internal: StashBuffer[InternalProtocol] = _
+
+  def stash(settings: EventsourcedSettings): StashBuffer[InternalProtocol] = {
+    if (internal == null) internal = StashBuffer(settings.stashCapacity)
```

Comment:
can use `akka.util.OptionVal` instead of `null`, it's allocation free, and match is also allocation free

Reply:
Fixed